test/cpp/end2end/client_fork_test.cc

```diff
@@ -47,6 +50,76 @@ int main(int /* argc */, char** /* argv */) { return 0; }
 namespace grpc::testing {
 namespace {
 
+class HangWatcher {
```

Comment:
Why is this better than just logging right before you do each step?  This seems like a lot more machinery than we need...

Reply:
As per our offline discussion, let's remove this before merging.